docs/planning/otel/specs/OTelWebSdk-Architecture.md

```diff
@@ -0,0 +1,2068 @@
+5. **Platform Detection**: Automatic platform-specific loading
+
+## Configuration Management Architecture
+
+The SDK implements a sophisticated configuration system designed for flexibility, validation, and runtime adaptability:
+
+### Configuration Hierarchy
+
+```typescript
+interface IOTelWebSdkConfig {
+  // Connection and endpoint configuration
+  connectionString?: string;
+  instrumentationKey?: string;  // Legacy support
+  endpoint?: string;
+  
+  // Resource identification
+  resource?: IResourceConfig;
+  
+  // Provider-specific configuration
+  tracing?: ITraceProviderConfig;
+  logging?: ILoggerProviderConfig;
+  
+  // Global settings
+  disabled?: boolean;
+  debug?: boolean;
+  environment?: string;
+  
+  // Performance and limits
+  performance?: IPerformanceConfig;
+  limits?: ILimitsConfig;
+  
+  // Extension points
+  plugins?: IPluginConfig[];
+  instrumentations?: IInstrumentationConfig[];
+  
+  // Runtime behavior
+  shutdownTimeout?: number;
+  flushTimeout?: number;
+  enableAutoInstrumentation?: boolean;
+}
+```
+
+### Configuration Principles
+
+1. **Hierarchical Override**: Configuration flows from global → provider → component levels
+2. **Runtime Validation**: All configuration validated at creation time with detailed error messages
+3. **Hot Reloading**: Support for runtime configuration updates where safe
+4. **Environment Detection**: Automatic detection of browser, framework, and deployment environment
+5. **Secure Defaults**: All defaults chosen for security and performance
+6. **Schema Validation**: JSON Schema validation for configuration objects
+
+### Configuration Sources
+
+```typescript
+// Priority order (highest to lowest):
+// 1. Explicit configuration passed to factory functions
+// 2. Environment variables (when available)
+// 3. Configuration providers (remote config, etc.)
+// 4. Default configuration
+
+const configManager = createConfigManager({
+  sources: [
+    new ExplicitConfigSource(userConfig),
+    new EnvironmentConfigSource(),
+    new RemoteConfigSource(configEndpoint),
+    new DefaultConfigSource()
+  ],
+  validation: {
+    strict: true,
+    throwOnError: true,
+    logValidationErrors: true
+  }
+});
+```
+
+## Error Handling and Resilience Architecture
+
+The SDK implements comprehensive error handling to ensure reliability in production environments:
+
+### Error Categories and Handling
+
+1. **Configuration Errors**: Validation failures, invalid settings
+   - Throw immediately during initialization
+   - Provide detailed error messages with correction suggestions
+   - Include configuration path and expected values
+
+2. **Network Errors**: Export failures, connectivity issues
+   - Implement exponential backoff with jitter
+   - Circuit breaker pattern for persistent failures
+   - Fallback to local storage when possible
+
+3. **Runtime Errors**: Unexpected exceptions during telemetry collection
+   - Graceful degradation to no-op behavior
```

Comment:
In the browser world a "no-op" behavior means do NOTHING -- return null / undefined, and NOT return a full no-op implementation of the interface.

There will be no "included" no-op implementation (unlike open telemetry) all callers MUST check that something was returned, not just explicitly use the returned "instance".

We may "provide" a loadable (separate) Sdk instance which is a No-Op instance (as part of the graceful handling of older browsers) from the CDN, but it SHOULD not be included within the SDK, ie. it MUST be tree-shakable.